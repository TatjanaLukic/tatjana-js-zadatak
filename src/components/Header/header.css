.header {
  display: flex;
  flex-direction: column;
  border: 2px solid #8e0a8e;
}
@media screen and (min-width: 576px) {
  .header {
    flex-direction: row;
    justify-content: space-between;
  }
}
@media screen and (min-width: 992px) {
  .header {
    width: 80%;
    margin: auto;
  }
}

/*# sourceMappingURL=header.css.map */
