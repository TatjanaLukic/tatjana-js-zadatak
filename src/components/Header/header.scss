@import "../../assets/varaiables";

.header {
  display: flex;
  flex-direction: column;

  @media screen and (min-width: $smallSize) {
    flex-direction: row;
    justify-content: space-between;
  }
}
