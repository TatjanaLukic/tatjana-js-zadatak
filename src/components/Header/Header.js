import React from "react";
import Navbar from "../Navbar/Navbar";
import CountryFilter from "../CountryFilter/CountryFilter";
import "./header.css";

const Header = () => {
  return (
      <header className="header">
          <Navbar />
          <CountryFilter />
      </header>
  )
};

export default Header;
