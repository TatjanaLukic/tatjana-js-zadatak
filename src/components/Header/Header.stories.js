import React from "react";
import Header from "./Header";
import { MemoryRouter } from "react-router-dom";

export default {
  title: "App/Header",
  component: Header,
  decorators: [(getStory) => <MemoryRouter>{getStory()}</MemoryRouter>]
};

const Template = () => <Header />;

export const Default = Template.bind({});