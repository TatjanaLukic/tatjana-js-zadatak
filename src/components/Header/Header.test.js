import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "../Header/Header";

describe("<Header />", () => {
  it("should render Navbar and CountryFilter", () => {
    render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    expect(screen.getByTestId("navbar")).toBeInTheDocument();
    expect(screen.getByTestId("country-filter")).toBeInTheDocument();
  });

  it("spanshots matche", () => {
    const { container } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByText("Categories"));
    expect(container).toMatchSnapshot();

  });
});
