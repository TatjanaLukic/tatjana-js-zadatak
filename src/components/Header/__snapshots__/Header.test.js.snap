// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> spanshots matche 1`] = `
<div>
  <header
    class="header"
  >
    <ul
      class="navbar-items"
      data-testid="navbar"
    >
      <li>
        <a
          class="navbar-item"
          href="/TopNewsPage"
        >
          Top News
        </a>
      </li>
      <li>
        <a
          class="navbar-item"
          href="/CategoriesPage"
        >
          Categories
        </a>
      </li>
      <li>
        <a
          class="navbar-item"
          href="/SearchPage"
        >
          Search
        </a>
      </li>
    </ul>
    <div
      class="country-filter"
      data-testid="country-filter"
    >
      <div
        data-testid="cf-item"
      >
        <input
          checked=""
          hidden=""
          id="us"
          type="radio"
          value="us"
        />
        <label
          class="country-filter-item country-filter-item--checked"
          for="us"
        >
          US
        </label>
      </div>
      <div
        data-testid="cf-item"
      >
        <input
          hidden=""
          id="gb"
          type="radio"
          value="gb"
        />
        <label
          class="country-filter-item"
          for="gb"
        >
          GB
        </label>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> spanshots matche 2`] = `
<div>
  <header
    class="header"
  >
    <ul
      class="navbar-items"
      data-testid="navbar"
    >
      <li>
        <a
          class="navbar-item"
          href="/TopNewsPage"
        >
          Top News
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="navbar-item navbar-item--checked"
          href="/CategoriesPage"
        >
          Categories
        </a>
      </li>
      <li>
        <a
          class="navbar-item"
          href="/SearchPage"
        >
          Search
        </a>
      </li>
    </ul>
    <div
      class="country-filter"
      data-testid="country-filter"
    >
      <div
        data-testid="cf-item"
      >
        <input
          checked=""
          hidden=""
          id="us"
          type="radio"
          value="us"
        />
        <label
          class="country-filter-item country-filter-item--checked"
          for="us"
        >
          US
        </label>
      </div>
      <div
        data-testid="cf-item"
      >
        <input
          hidden=""
          id="gb"
          type="radio"
          value="gb"
        />
        <label
          class="country-filter-item"
          for="gb"
        >
          GB
        </label>
      </div>
    </div>
  </header>
</div>
`;
